### ISSUE TITLE
Swagger pyang plugin fails when "uses" statement is invoked inside a "grouping" statement

### ISSUE THREAD

**Issue by grigio60** [ict-strauss/COP#1]
Hello

I have a set of yang modules where some basic and commonly used information is defined centrally as a grouping. This grouping is then recalled by other yang definitions through the "uses" statement.

According to RFC6020, the "uses" statement can be used a sub-statement also inside a "grouping" statement, so allowing a flexible reuse. However, the swagger plugin aborts with an error when this is applied, although pyang validates the code without any errors nor warnings.

The sample code to reproduce the issue is split into different files:
1. basic.yang includes the definition of the basic grouping
2. complex.yang includes the "uses" of the basic grouping inside another "container"
3. complex2.yang includes the "uses" of the basic grouping inside another "groupin".

Pyang validates successfully all three source files without error.
1. _pyang basic.yang -f swagger -o basic.json_   executes successfully without errors
2. _pyang complex.yang -f swagger -o complex.json_  executes successfully without errors
3. _pyang complex2.yang .f swagger -o complex2.json_ fails with the following messages:

> Traceback (most recent call last):
>  File "c:/Python34/Scripts/pyang", line 403, in <module> run
>   File "c:/Python34/Scripts/pyang", line 377, in run emit_obj.emit(ctx, modules, fd)
>   File "c:\Python34\lib\site-packages\pyang/plugins\swagger.py", line 71, in emit 
>     emit_swagger_spec(ctx, modules, fd, ctx.opts.path)
>   File "c:\Python34\lib\site-packages\pyang/plugins\swagger.py", line 126, in emit_swagger_spec
>     gen_model(pending_models, definitions)
>   File "c:\Python34\lib\site-packages\pyang/plugins\swagger.py", line 126, in gen_model
>     PARENT_MODELS[ref]['models'].append(child.arg)
> KeyError: 'CommonAttributes'

Notice that "commonAttributes" is exactly the name of the grouping invoked inside another grouping (I tested the same sequence using a different name and the KeyError always returned consistently that name).

Removing all the "uses" statements inside all "grouping" definitions is not a viable work-around for real projects, especially as long as IETF- or other standard-based yang modules are involved.

Additional information:
Python version (python -V): Python 3.4.3
Pyang version (pyang -v): pyang 1.5

Attached the contents of the files used to reproduce the issue.

basic.yang:

``` yang
module basic {
    namespace "urn:acme:yang:basic";
    prefix "basic";

    grouping commonAttributes {
        leaf my-id {
            type uint8;
        }
        leaf my-name {
            type string;
        }
        leaf my-status {
            type boolean;
            config false;
        }
    }
}
```

complex.yang:

``` yang
module complex {
    namespace "urn:acme:yang:complex";
    prefix "complex";

    import basic {
        prefix "basic";
    }

    grouping myClass {
        leaf alpha {
            type string;
        }
        leaf beta {
            type uint8;
        }
    }

    container myNode {
        list subNodes {
            key "id";
            leaf id {
                type uint8;
            }
            uses myClass;
        }
        container myChild {
            uses basic:commonAttributes;
            leaf gamma {
                type boolean;
            }
            leaf delta {
                type string;
            }
        }
    }
}
```

complex2.yang:

``` yang
    module complex2 {
        namespace "urn:acme:yang:complex2";
        prefix "complex";

        import basic {
            prefix "basic";
        }

        grouping myClass {
            uses basic:commonAttributes;       // OFFENDING STATEMENT
            leaf alpha {
                type string;
            }
            leaf beta {
                type uint8;
            }
        }

        container myNode {
            list subNodes {
                key "id";
                leaf id {
                    type uint8;
                }
                uses myClass;
            }
            container myChild {
                leaf gamma {
                    type boolean;
                }
                leaf delta {
                    type string;
                }
            }
        }
    }
```

**Comment by grigio60** [comment]
Additional comment: 

The same plugin error occurs also if all the definitions are enclosed within the same single module.
I got the same error results when submitting this sample yang code:

``` yang
module basic {
    namespace "urn:acme:yang:basic";
    prefix "basic";

    grouping commonAttributes {
        leaf my-id {
            type uint8;
        }
        leaf my-name {
            type string;
        }
        leaf my-status {
            type boolean;
            config false;
        }
    }
    grouping myClass {
        uses commonAttributes;    // OFFENDING STATEMENT
        leaf alpha {
            type string;
        }
        leaf beta {
            type uint8;
        }
    }
}
```

**Comment by arthurMll** [comment]
Hello grigio60,

I update the code in order to solve the issues you mentioned, apologies for the delay but it was not a straightforward fix S:

The first problem I encountered was that the uses statement inside a grouping were not able to reference imported models, I fix that but then other bugs emerged. The main problems was related with the dependency to use grouping statements to define the data structures used by the tree-nodes (container and list statements).

I verified the new version with the models you provide us and it works. In any case, feel free to generate the code by your own and if you find any bug or error don't doubt to contact us.

Thanks you very much for you feedback,
Arturo.

**Comment by grigio60** [comment]
Arturo,

Many thanks for this quick intervention.

Tested the new code of swagger plugin with my original files, the fix is working :)
Now I'm able to get the swagger json from almost all of my yang definitions.

Actually I'm still blocked on swagger plugin on a couple of yang files but this is now due only to the second blocking issue I've flagged, not to this one. This first fix it's a big step ahead in any case.

Thank you very much again for this good job,
Germano
